File: tournaments/arrayMode/arrayMode.pl
#!/usr/bin/perl
use strict;
use warnings FATAL => 'all';

sub arrayMode {
    my ($sequence) = @_;

    my %store = ();
    my $result = -1;
    my $count = 0;

    foreach my $item (@$sequence) {
        if ($store{$item}) {
            ++$store{$item};
        } else {
            $store{$item} = 1;
        }
    }

    foreach my $key (keys(%store)) {
        if ($store{$key} > $count) {
            $count = $store{$key};
            $result = int($key);
        }
    }

    return $result;
}

File: tournaments/arrayMode/arrayMode.js
function arrayMode(sequence) {
    const store = new Map();
    let result = -1;
    let count = 0;
    
    for (const item of sequence) {
        if (store.has(item)) {
            store.set(item, 1 + store.get(item));
        } else {
            store.set(item, 1);
        }
    }
    
    for (const [key, value] of store) {
        if (count < value) {
            count = value;
            result = key;
        }
    }
    
    return result;
}
